### DISCUSSION TITLE
Even with no clients pgbouncer shows active connections

### DISCUSSION THREAD

**Discussion by radiantone** [pgbouncer/pgbouncer#754] [Q&A]
Hi,
  When my client connects (SQLAlchemy app), about 10 connections get opened. I can see all the clients in "show clients". When I quit my app, the clients disappear, but pgb still shows 10 active connections. Even though I close the connections in client as well.

What is going on here?

thanks!
D

**Comment by petere** [comment]
If the app quits without closing the connections, pgbouncer will still see the connections until the kernel kills them after the TCP timeout mechanism.

**Reply by radiantone** [reply]
The app does close the connections. Pgbouncer need to garbage collect its own connections if there is no client on the other end. This is easily dont via the socket handling. When a client closes its end (which happens automatically when a process exits) pgbouncer can easily detect this. But it doesnt.

**Reply by petere** [reply]
You can tune the TCP keepalive settings in `pgbouncer.ini` to control this "garbage collection".
